File: examples/nodejs/detect_backup_file_missing.js
#!/usr/bin/env node
/**
 * Detect Backup File Missing
 * Problem: Backup job completes but backup file is missing
 * Solution: Verify backup file exists after backup creation
 * Documentation: https://deadmanping.com/detect-backup-file-missing
 */

const fs = require('fs');
const { execSync } = require('child_process');
const https = require('https');

const MONITOR_ID = 'YOUR_MONITOR_ID';  // Replace with your actual monitor ID
const backupFile = '/backups/db-backup.sql.gz';

// Run backup
execSync('pg_dump mydb | gzip > ' + backupFile);

// Get file data
let fileExists = fs.existsSync(backupFile);
let fileSize = 0;
if (fileExists) {
  fileSize = fs.statSync(backupFile).size;
}

// Always ping with file data in payload
// In DeadManPing panel: set validation rules:
//   - "file_exists" == true
//   - "size" > 0
// Panel will automatically detect if file is missing or empty
https.request(`https://deadmanping.com/api/ping/${MONITOR_ID}?file_exists=${fileExists}&size=${fileSize}`, { method: 'POST' }).end();

File: examples/python/detect_backup_file_missing.py
#!/usr/bin/env python3
"""
Detect Backup File Missing
Problem: Backup job completes but backup file is missing
Solution: Verify backup file exists after backup creation
Documentation: https://deadmanping.com/detect-backup-file-missing
"""

import os
import subprocess
import requests
import sys

MONITOR_ID = "YOUR_MONITOR_ID"  # Replace with your actual monitor ID
backup_file = "/backups/db-backup.sql.gz"

# Run backup
subprocess.run(["pg_dump", "mydb"], stdout=open(backup_file.replace('.gz', ''), "w"))
subprocess.run(["gzip", backup_file.replace('.gz', '')])

# Get file data
file_exists = os.path.exists(backup_file)
file_size = os.path.getsize(backup_file) if file_exists else 0

# Always ping with file data in payload
# In DeadManPing panel: set validation rules:
#   - "file_exists" == True
#   - "size" > 0
# Panel will automatically detect if file is missing or empty
requests.post(f"https://deadmanping.com/api/ping/{MONITOR_ID}?file_exists={file_exists}&size={file_size}")
